### DISCUSSION TITLE
2-Gang Wall Switch - How to use 1 button for both relays?

### DISCUSSION THREAD

**Discussion by grillp** [arendst/Tasmota#11666] [Q&A] [answered]
(Cross-post from r/tasmota as I found this discussion group later)
Hey all,

I have one of these 2 Gang Switches: https://templates.blakadder.com/deta_6912HA.html

I have successfully flashed with the latest Tasmota (Own compile: MQTT TLS, WEB, Rules, HA) and configured using the template from the link above. Connected to HA using MQTT (with autoconfig SetOption19).

I have the 2 relays wired up to 2 sets of light. As both sets of light always come on together, I want to use one of the buttons on the device (e.g Button1) to trigger both relays (Power1 and Power2). Then I can use the other button in HA for other purposes.

Also, I want the default behaviour of the device (if not connected to HA/MQTT for whatever reason) to remain for that 1 switch to control both relays.

(I know the easiest thing would be to wire both lights to the same relay in parallel, which I am considering.. but want to try via config first)

After checking online, I could not find exactly what I was looking for, but the gist of it seemed to be

1. Decouple the switches from the relays (`SetOption73 1`)
2. Create a rule that would toggle both Power1 and Power2 on Switch1#State=2 (`Rule1 ON Button2#State=2 DO Backlog POWER1 TOGGLE; POWER2 TOGGLE ENDON`)

So I tried that, but pressing buttons did nothing.

Next, I disabled Option73 (`SO73 0`). Then Swith1 did switch both relays, but Switch2 also turned off Power2, which is not what I wanted (so the toggles from Switch1 means the 2 Powers were out of sync).

Any idea what I can do (except rewiring everything into 1 switch)?

Cheers!

G./

**Comment by barbudor** [comment, marked as answer]
1) You first need to decide which method works the best for the button you want to use in HA as you can;
- Use SetOption73
- Use ButtonTopic
- Use rule with a customer publish topic
I know nothing about HA so I can't advise you here
2) Depending on your choice for 1), a rule for the button that will control the local relay will be needed. Exact trigger depends on 1)

**Reply by grillp** [reply]
Hey @barbudor , thanks for your reply.

So do you mean I need to do ALL of the 3 things you mention in 1) ?

Or just one of the 3?

I can already see messages appearing in HA that I can use for the second button, so that is cool.

But if I set option 73, the Rule1 (as defined above) does not work. Is the Rule waiting for the wrong thing? I.e. with Op73 does the  Button2#State=2 not fire?

Still confused.

Thanks

G./

**Comment by haze1986** [comment]
Hi can I check if you have your answer to this? Hope to bump this up to test the solution, thanks.
